backstop/src/emissions/manager.rs

```diff
@@ -39,7 +41,69 @@ pub fn add_to_reward_zone(e: &Env, to_add: Address, to_remove: Address) {
+            Some(to_remove) => {
+                // swap to_add for to_remove
+                let to_remove_index = reward_zone.first_index_of(to_remove.clone());
+                match to_remove_index {
+                    Some(idx) => {
+                        // verify distribute was run recently to prevent "to_remove" from losing excess emissions
+                        // @dev: resource constraints prevent us from distributing on reward zone changes
+                        let last_distribution = storage::get_last_distribution_time(e);
+                        if last_distribution < e.ledger().timestamp() - 24 * 60 * 60 {
+                            panic_with_error!(e, BackstopError::BadRequest);
+                        }
+
+                        // Verify "to_add" has a higher backstop deposit that "to_remove"
+                        if pool_data.tokens <= storage::get_pool_balance(e, &to_remove).tokens {
+                            panic_with_error!(e, BackstopError::InvalidRewardZoneEntry);
+                        }
+
+                        // update backstop emissions for the pool before removing it from the reward zone
+                        // set emission index to i128::MAX to prevent further emissions
+                        let to_remove_emis_data =
+                            storage::get_rz_emis_data(e, &to_remove).unwrap_optimized();
+                        set_rz_emissions(
+                            e,
+                            &to_remove,
+                            i128::MAX,
+                            to_remove_emis_data.accrued,
+                            false,
+                        );
+
+                        reward_zone.set(idx, to_add.clone());
+                    }
+                    None => panic_with_error!(e, BackstopError::InvalidRewardZoneEntry),
+                }
+            }
+        }
+    }
+    // Set the new pool's backstop emissions index to the current gulp index
+    if let Some(to_add_emis_data) = storage::get_rz_emis_data(e, &to_add) {
+        set_rz_emissions(
+            e,
+            &to_add,
+            rz_emission_index,
+            to_add_emis_data.accrued,
+            false,
+        );
+    } else {
+        set_rz_emissions(e, &to_add, rz_emission_index, 0, false);
+    }
+    storage::set_reward_zone(e, &reward_zone);
+}
+
+/// remove a pool to the reward zone if below the minimum backstop deposit threshold
+pub fn remove_from_reward_zone(e: &Env, to_remove: Address) {
+    let mut reward_zone = storage::get_reward_zone(e);
+
+    // enusre to_add has met the minimum backstop deposit threshold
+    // NOTE: "to_add" can only carry a pool balance if it is a deployed pool from the factory
+    let pool_data = load_pool_backstop_data(e, &to_remove);
+    if require_pool_above_threshold(&pool_data) {
+        panic_with_error!(e, BackstopError::BadRequest);
+    } else {
         let to_remove_index = reward_zone.first_index_of(to_remove.clone());
```

Comment:
can the core logic of `remove_from_reward_zone` be shared?

IE

```rust
/// Remove pool from the reward zone vec, does not store
fn remove_pool(reward_zone: &mut Vec<Address>, to_remove: &Address) {
    // check for existance of pool in the rz
    // check last distro time was recent
   // update rx_emissions
  // remove from rz_vec
}
```

Then both `add_to_rz` and `remove_from_rz` can share the most important chunk of logic when removing a pool.